src/Microsoft.AspNetCore.Mvc.Core/Internal/ResponseCacheFilter.cs

```diff
@@ -110,6 +121,16 @@ public void OnActionExecuting(ActionExecutingContext context)
                 headers[HeaderNames.Vary] = VaryByHeader;
             }
 
+            if (VaryByQueryKeys != null)
+            {
+                var responseCacheFeature = context.HttpContext.Features.Get<IResponseCacheFeature>();
+                if (responseCacheFeature == null)
+                {
+                    throw new InvalidOperationException($"The response cache middleware must be added when using the {nameof(VaryByQueryKeys)} feature.");
+                }
+                responseCacheFeature.VaryByQueryKeys = VaryByQueryKeys;
```

Comment:
That might work although I'm not sure if the current set of options on the attribute lets you configure server \ client side caching with a good level of fidelity. (i.e. enable response caching on this action but not server caching etc) which `OutputCacheAttribute` allowed via https://msdn.microsoft.com/en-us/library/system.web.ui.outputcachelocation(v=vs.110).aspx. Plus I assumed we specifically called it response caching ) to differentiate it from output caching (which was on the server side).